src/servico/AdministracaoDeUsuariosService.java

```diff
@@ -38,15 +38,16 @@ public void SalvarUsuario(String Nome, String Senha, Perfil profile){
     }
     
     public void EditarUsuario(String senha, Perfil profile, Long id) throws NonexistentEntityException, Exception{
-        
-        UsuarioJpaController controller = new UsuarioJpaController(emf);
-        Usuario usuarioAEditar = controller.findUsuario(id);
-        
-        usuarioAEditar.setUsuario(usuarioAEditar.getUsuario());
-        usuarioAEditar.setSenha(senha);
-        usuarioAEditar.setPermissao(profile);
-        
-        controller.edit(usuarioAEditar);
+        if(!senha.equals("")){

```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/calebepb/substituicao/file/2769975817/issues/source?bid=3308597&fileBranchId=3353677#l41)